src/getSearchKeywords.js

```diff
@@ -0,0 +1,53 @@
+function getSearchKeywords() {
+  const currentUrl = new URL(window.location.href);
+  const urlParams = new URLSearchParams(currentUrl.search);
+  const allCondition = ["+", " AND ", "|", " OR ", " -", `"`, " ", ","];
+  let keywords;
+
+  for (const key of urlParams.keys()) {
+    if (key === "q" || key === "query") {
+      keywords = urlParams.get(key);
+    }
+  }
+
+  if (keywords.includes(`"`)) {
+    keywords = keywords.replaceAll(`"`, "");
+    return [keywords];
+  }
+
+  for (let i = 0; i < allCondition.length; i++) {
+    if (keywords === allCondition[i]) {
+      return [keywords];
+    }
+  }
+
+  for (let i = 0; i < allCondition.length; i++) {
+    if (
+      keywords[0] === allCondition[i] ||
+      keywords[keywords.length - 1] === allCondition[i]
+    ) {
+      return [keywords];
+    }
+  }
+
+  if (keywords.includes(" -")) {
+    const deleteIndex = keywords.indexOf(" -");
+    keywords = keywords.slice(0, deleteIndex);
+  }
+
+  allCondition.forEach((condition) => {
+    if (keywords.includes(condition)) {
+      keywords = keywords.split(condition);
+    }
+  });
+
+  if (typeof keywords === "object") {
```

Comment:
typeof null 도 결과물이 'object'로 나오더라구요. 혹시 배열인 지 확인하고 싶으셨던 것이라면 다음의 방법은 어떨까요? 딱 배열인지만 확인해주더라구요!
```suggestion
  if (Array.isArray(keywords)) {
```

Reply:
감사합니다. 의견 반영하여 수정하겠습니다~